src/dovi/rpu_info.rs

```diff
@@ -333,17 +380,242 @@ impl RpusListSummary {
             rpu_mastering_meta_str,
             profiles_str,
             dm_version_str,
             dm_version_counts,
+            dmv2,
             l6_meta,
+            l5_str,
+            l8_trims,
+            l9_mdp,
             l1_data,
             l1_stats,
             l2_trims,
+            l2_data: None,
+            l2_stats: None,
+            l8_data: None,
+            l8_stats_trims: None,
+            l8_stats_saturation: None,
+            l8_stats_hue: None,
         })
     }
+
+    pub fn with_l2_data(rpus: &[DoviRpu]) -> Result<Self> {
+        let mut summary = Self::new(rpus)?;
+
+        let l2_data: Vec<_> = rpus
+            .iter()
+            .map(|rpu| {
+                if let Some(ExtMetadataBlock::Level2(l2)) =
+                    rpu.vdr_dm_data.as_ref().and_then(|dm| dm.get_block(2))
+                {
+                    l2.clone()
+                } else {
+                    ExtMetadataBlockLevel2::default()
+                }
+            })
+            .collect();
+
+        summary.l2_stats = Some(SummaryTrimsStats {
+            slope: Self::min_max_avg(&l2_data, |e| e.trim_slope as f64),
+            offset: Self::min_max_avg(&l2_data, |e| e.trim_offset as f64),
+            power: Self::min_max_avg(&l2_data, |e| e.trim_power as f64),
+            chroma: Self::min_max_avg(&l2_data, |e| e.trim_chroma_weight as f64),
+            saturation: Self::min_max_avg(&l2_data, |e| e.trim_saturation_gain as f64),
+            ms_weight: Self::min_max_avg(&l2_data, |e| e.ms_weight as f64),
+        });
+        summary.l2_data = Some(l2_data);
+
+        Ok(summary)
+    }
+
+    fn with_l8_data(rpus: &[DoviRpu]) -> Result<Self> {
+        let mut summary = Self::new(rpus)?;
+
+        let l8_data: Vec<_> = rpus
+            .iter()
+            .map(|rpu| {
+                if let Some(ExtMetadataBlock::Level8(l8)) =
+                    rpu.vdr_dm_data.as_ref().and_then(|dm| dm.get_block(8))
+                {
+                    l8.clone()
+                } else {
+                    ExtMetadataBlockLevel8::default()
+                }
+            })
+            .collect();
+
+        summary.l8_data = Some(l8_data);
+        Ok(summary)
+    }
+
+    pub fn with_l8_trims_data(rpus: &[DoviRpu]) -> Result<Self> {
+        let mut summary = Self::with_l8_data(rpus)?;
+
+        if let Some(l8_data) = summary.l8_data.as_ref() {
+            summary.l8_stats_trims = Some(SummaryTrimsStats {
+                slope: Self::min_max_avg(l8_data, |e| e.trim_slope as f64),
+                offset: Self::min_max_avg(l8_data, |e| e.trim_offset as f64),
+                power: Self::min_max_avg(l8_data, |e| e.trim_power as f64),
+                chroma: Self::min_max_avg(l8_data, |e| e.trim_chroma_weight as f64),
+                saturation: Self::min_max_avg(l8_data, |e| e.trim_saturation_gain as f64),
+                ms_weight: Self::min_max_avg(l8_data, |e| e.ms_weight as f64),
```

Comment:
Probably should include `target_mid_contrast` and `clip_trim` for L8.
Would be `Option`s  in the same trims struct.